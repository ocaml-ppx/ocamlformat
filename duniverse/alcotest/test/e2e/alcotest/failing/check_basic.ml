(* Check that v of type [typ] matches with itself *)
let id_case typ typ_str v1 v2 =
  Alcotest.test_case typ_str `Quick (fun () -> Alcotest.check typ typ_str v1 v2)

let () =
  let open Alcotest in
  run ~verbose:true __FILE__
    [
      ( "different basic",
        [
          id_case bool "bool" true false;
          id_case int "int" 1 2;
          id_case int32 "int32" Int32.max_int Int32.min_int;
          id_case int64 "int64" Int64.max_int Int64.min_int;
          id_case (float 0.0) "float" 1.0 2.0;
          id_case char "char" 'a' 'b';
          id_case string "string" "Lorem ipsum" "dolor sit amet.";
          id_case bytes "bytes"
            (Bytes.of_string "\x01\x02\x03")
            (Bytes.of_string "\x01\x00\x03");
        ] );
      ( "different composite",
        [
          id_case (list char) "list" [ 'a'; 'b' ] [ 'a'; 'c' ];
          id_case (array int64) "array"
            Int64.[| zero; max_int |]
            Int64.[| zero; one; max_int |];
          id_case (option int) "option some" (Some 1) (Some 2);
          id_case (result int unit) "result" (Ok 1) (Error ());
          id_case (pair int char) "pair" (1, 'a') (1, 'b');
          id_case (triple int bool string) "triple" (1, true, "a")
            (1, false, "a");
        ] );
    ]
