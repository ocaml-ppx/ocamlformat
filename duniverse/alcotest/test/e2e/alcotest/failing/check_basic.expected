Testing `test/e2e/alcotest/failing/check_basic.ml'.
This run has ID `<uuid>'.

ASSERT bool
  [FAIL]        different basic              0   bool.
ASSERT int
  [FAIL]        different basic              1   int.
ASSERT int32
  [FAIL]        different basic              2   int32.
ASSERT int64
  [FAIL]        different basic              3   int64.
ASSERT float
  [FAIL]        different basic              4   float.
ASSERT char
  [FAIL]        different basic              5   char.
ASSERT string
  [FAIL]        different basic              6   string.
ASSERT bytes
  [FAIL]        different basic              7   bytes.
ASSERT list
  [FAIL]        different composite          0   list.
ASSERT array
  [FAIL]        different composite          1   array.
ASSERT option some
  [FAIL]        different composite          2   option some.
ASSERT result
  [FAIL]        different composite          3   result.
ASSERT pair
  [FAIL]        different composite          4   pair.
ASSERT triple
  [FAIL]        different composite          5   triple.

┌──────────────────────────────────────────────────────────────────────────────┐
│ [FAIL]        different basic              0   bool.                         │
└──────────────────────────────────────────────────────────────────────────────┘
FAIL bool

   Expected: `true'
   Received: `false'
 ──────────────────────────────────────────────────────────────────────────────


┌──────────────────────────────────────────────────────────────────────────────┐
│ [FAIL]        different basic              1   int.                          │
└──────────────────────────────────────────────────────────────────────────────┘
FAIL int

   Expected: `1'
   Received: `2'
 ──────────────────────────────────────────────────────────────────────────────


┌──────────────────────────────────────────────────────────────────────────────┐
│ [FAIL]        different basic              2   int32.                        │
└──────────────────────────────────────────────────────────────────────────────┘
FAIL int32

   Expected: `2147483647'
   Received: `-2147483648'
 ──────────────────────────────────────────────────────────────────────────────


┌──────────────────────────────────────────────────────────────────────────────┐
│ [FAIL]        different basic              3   int64.                        │
└──────────────────────────────────────────────────────────────────────────────┘
FAIL int64

   Expected: `9223372036854775807'
   Received: `-9223372036854775808'
 ──────────────────────────────────────────────────────────────────────────────


┌──────────────────────────────────────────────────────────────────────────────┐
│ [FAIL]        different basic              4   float.                        │
└──────────────────────────────────────────────────────────────────────────────┘
FAIL float

   Expected: `1'
   Received: `2'
 ──────────────────────────────────────────────────────────────────────────────


┌──────────────────────────────────────────────────────────────────────────────┐
│ [FAIL]        different basic              5   char.                         │
└──────────────────────────────────────────────────────────────────────────────┘
FAIL char

   Expected: `a'
   Received: `b'
 ──────────────────────────────────────────────────────────────────────────────


┌──────────────────────────────────────────────────────────────────────────────┐
│ [FAIL]        different basic              6   string.                       │
└──────────────────────────────────────────────────────────────────────────────┘
FAIL string

   Expected: `Lorem ipsum'
   Received: `dolor sit amet.'
 ──────────────────────────────────────────────────────────────────────────────


┌──────────────────────────────────────────────────────────────────────────────┐
│ [FAIL]        different basic              7   bytes.                        │
└──────────────────────────────────────────────────────────────────────────────┘
FAIL bytes

   Expected: `"\001\002\003"'
   Received: `"\001\000\003"'
 ──────────────────────────────────────────────────────────────────────────────


┌──────────────────────────────────────────────────────────────────────────────┐
│ [FAIL]        different composite          0   list.                         │
└──────────────────────────────────────────────────────────────────────────────┘
FAIL list

   Expected: `[a; b]'
   Received: `[a; c]'
 ──────────────────────────────────────────────────────────────────────────────


┌──────────────────────────────────────────────────────────────────────────────┐
│ [FAIL]        different composite          1   array.                        │
└──────────────────────────────────────────────────────────────────────────────┘
FAIL array

   Expected: `[|0; 9223372036854775807|]'
   Received: `[|0; 1; 9223372036854775807|]'
 ──────────────────────────────────────────────────────────────────────────────


┌──────────────────────────────────────────────────────────────────────────────┐
│ [FAIL]        different composite          2   option some.                  │
└──────────────────────────────────────────────────────────────────────────────┘
FAIL option some

   Expected: `Some 1'
   Received: `Some 2'
 ──────────────────────────────────────────────────────────────────────────────


┌──────────────────────────────────────────────────────────────────────────────┐
│ [FAIL]        different composite          3   result.                       │
└──────────────────────────────────────────────────────────────────────────────┘
FAIL result

   Expected: `Ok 1'
   Received: `Error ()'
 ──────────────────────────────────────────────────────────────────────────────


┌──────────────────────────────────────────────────────────────────────────────┐
│ [FAIL]        different composite          4   pair.                         │
└──────────────────────────────────────────────────────────────────────────────┘
FAIL pair

   Expected: `(1, a)'
   Received: `(1, b)'
 ──────────────────────────────────────────────────────────────────────────────


┌──────────────────────────────────────────────────────────────────────────────┐
│ [FAIL]        different composite          5   triple.                       │
└──────────────────────────────────────────────────────────────────────────────┘
FAIL triple

   Expected: `(1, true, a)'
   Received: `(1, false, a)'
 ──────────────────────────────────────────────────────────────────────────────

14 failures! in <test-duration>s. 14 tests run.
